src/Sql/Sql/Elastic Jobs/Cmdlet/Agent/RemoveAzureSqlElasticJobAgent.cs

```diff
@@ -0,0 +1,167 @@
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.ElasticJobs.Cmdlet
+{
+    /// <summary>
+    /// Defines the Remove-AzureRmSqlElasticJobAgent cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlElasticJobAgent",
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = DefaultParameterSet)]
+    [OutputType(typeof(AzureSqlElasticJobAgentModel))]
+    public class RemoveAzureSqlElasticJobAgent : AzureSqlElasticJobAgentCmdletBase<AzureSqlElasticJobAgentModel>
+    {
+        /// <summary>
+        /// Gets or sets the resource group name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 0,
+            HelpMessage = "The resource group name")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agent input object model
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The agent object")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlElasticJobAgentModel InputObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agent resource id
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The agent resource id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the server name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 1,
+            HelpMessage = "The server name")]
+        [ValidateNotNullOrEmpty]
+        public override string ServerName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the agent to use.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 2,
+            HelpMessage = "The agent name")]
+        [ValidateNotNullOrEmpty]
+        [Alias("AgentName")]
+        public override string Name { get; set; }
+
+        /// <summary>
+        /// Defines whether it is ok to skip the requesting of rule removal confirmation
+        /// </summary>
+        [Parameter(HelpMessage = "Skip confirmation message for performing the action")]
+        public SwitchParameter Force { get; set; }
```

Comment:
@johnpaulkee do we need the `-Force` parameter here? It should only be needed in cases where we are prompting the user for confirmation and want to give them a way to bypass the prompt. For example, if deleting this agent would delete additional resources, then we would want to prompt the user to confirm they want to perform this action and notify them that additional resources would be deleted -- passing the `-Force` parameter would skip this prompt. Is this the desired behavior?

Reply:
Yup this is the desired behavior.
I'd also try not to remove the cmdlet since there are those already using the Az.Sql-preview release cmdlets and they may have automation that already include the flag.